cterasdk/edge/sync.py

```diff
@@ -78,6 +79,45 @@ def refresh(self):
         self._gateway.execute("/config/cloudsync/cloudExtender", "refreshPaths", None)
         logging.getLogger().info("Completed refreshing cloud folders.")
 
+    def exclude_files(self, extensions=None, filenames=None, paths=None, custom_exclusion_rules=None):
+        """
+        Exclude files from Cloud Sync. This method will override any existing file exclusion rules
+        Use :func:`cterasdk.edge.types.FileExclusionBuilder` to build custom file exclusion rules`
+
+        :param list[str] extensions: List of file extensions
+        :param list[str] filenames: List of file names
+        :param list[str] paths: List of file paths
+        :param list[cterasdk.common.types.FilterBackupSet] rules: Set of custom exclusion rules
+        """
+        rules = list()
+        if extensions:
+            param = FilterBackupSet('List of file extensions to exclude from sync',
+                                    filter_rules=[FileExclusionBuilder.extensions().include(extensions).build()])
+            rules.append(param)
+        if filenames:
+            param = FilterBackupSet('List of file names to exclude from sync',
+                                    filter_rules=[FileExclusionBuilder.names().include(filenames).build()])
+            rules.append(param)
+        if paths:
+            param = FilterBackupSet('List of file paths to exclude from sync',
+                                    filter_rules=[FileExclusionBuilder.paths().include(filenames).build()])
+            rules.append(param)
+        if custom_exclusion_rules:
+            rules.append(custom_exclusion_rules)
```

Comment:
I think this should be extend and not append. append will add the entire list as the next element in the rules list

Reply:
Good catch

Reply:
Using rules.extend(custom_exclusion_rules) is a better practice as it writes directly to "rules" instead of creating a temporary variable on the right side of the the assignment operation

Reply:
Done